### PR TITLE
feat: add drawbridge integration

### LINKED ISSUE #27: Drawbridge integration
It would be a nice improvement to automatically create a Drawbridge user, repository and tag on upload of files and pass a slug to that Drawbridge artifact to `enarx` rather than actual contents.

### PR DESCRIPTION
**Pull request by definitelynobody**
Closes #27

many of the included demos take a long time to start or have other issues, that's out of scope for this PR but I can remove them if required

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -385,23 +425,52 @@ async fn root_post(
...
-        .stdin(Stdio::null())
-        .stdout(Stdio::piped())
-        .stderr(Stdio::piped())
-        .kill_on_drop(true)
-        .spawn()
-        .map_err(|e| {
-            error!("failed to spawn process: {e}");
-            StatusCode::INTERNAL_SERVER_ERROR.into_response()
-        })?;
+    let exec = match workload_type.as_str() {
```
Comment (rvolosatovs) [inline src/main.rs:430]: We probably want an enum here

**Review by rvolosatovs — APPROVED** [review]

**Comment by rvolosatovs** [comment]
Since this is not ready for review and requires rebasing, marking as draft and removing approval for a repeated review

File: src/main.rs [hunk 2]
```diff
@@ -127,25 +144,40 @@ impl Args {
 
 #[derive(Copy, Clone, Debug)]
 struct Limits {
+    /// Size in megabytes
     size_limit_default: usize,
```
Comment (rvolosatovs) [inline src/main.rs:148]: I think these should be bytes. We should (eventually) provide functionality to be able to specify e.g. `10m` as a valid config value

File: src/main.rs [hunk 3]
```diff
@@ -19,23 +19,40 @@ use std::io::Write;
...
+
 // TODO: raise this when this is fixed: https://github.com/profianinc/benefice/issues/75
 const READ_TIMEOUT: Duration = Duration::from_millis(500);
 const TOML_MAX: usize = 256 * 1024; // 256 KiB
 
+lazy_static! {
+    static ref EXAMPLES: Vec<&'static str> = include_str!("../examples.txt")
+        .lines()
+        .filter(|line| !line.is_empty())
+        .collect::<Vec<_>>();
```
[the remark this answers is not in the document]
Comment (definitelynobody) [reply src/main.rs:53]: its easier to sort and add more examples

File: src/main.rs (hunk 2 above)
Comment (definitelynobody) [reply src/main.rs:148]: It used to be in bytes a few weeks ago but was changed to megabytes outside of this pr. This just documents what these fields already do. Also I can't really see a use case for taking kilobytes or gigabytes as an argument

File: src/main.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (rvolosatovs) [reply src/main.rs:53]: I disagree with this, but whatever works!

**Review by rvolosatovs — APPROVED** [review]

**Pull request merged**